### DISCUSSION TITLE
How to make a PR ? And am I following the right folder structure ?

### DISCUSSION THREAD

**Discussion by heysujal** [open-sauced-craftwork/browser-extension#8] [Q&A] [answered]
I started this project by cloning my template repository  [from here](https://github.com/open-sauced-craftwork/open_sauced_craftwork_browser_extension-heysujal) . Then I followed the instruction given in the readme. I wanted to make sure if the structure of the folders is okay or not because I tried to get everything at the root level but I was facing some errors.
The chrome extension folder is inside this root folder. 

![image](https://user-images.githubusercontent.com/55016909/224042048-1e089da4-af0e-40e8-9bdb-7c548462e5ad.png)

Also , I am not able to make a PR as I am not getting any option to make one. Is it because I cloned a wrong repository ?

**Comment by bdougie** [comment]
@heysujal the path to this is from a new branch. I see you pushed all your changes to main, and a PR is meant compare to main from your new branch. 

1. My suggest is create a new branch called `connect-github-to-open-sauced` and push that to GitHub with the same history. 
2. Next checkout the main branch and run `git reset HEAD~5` from the command line. I do not know the equivalent in VS Code, but you are a quick google search away from getting that answer. 

Let me know if that helps.

**Reply by heysujal** [reply]
It helped. Now, I have shifted all my changes from the main branch to another branch and created a PR. Looking forward to your feedback.

**Comment by ishaanbedi** [comment]
@bdougie I am facing a similar issue.

On the `main` branch, there's just one commit from the GitHub classroom bot that got generated when I participated in the task:
<img width="911" alt="Screenshot 2023-03-10 at 9 00 18 AM" src="https://user-images.githubusercontent.com/39641326/224216460-bdde8a2b-2926-4e12-b59f-8680050e67b9.png">

On my custom branch `opensauced-github-extension-ishaanbedi` , I've made all my commits:
<img width="906" alt="Screenshot 2023-03-10 at 9 00 36 AM" src="https://user-images.githubusercontent.com/39641326/224216977-35bdc80a-c9a0-47ca-9ed5-ad97f9886e0c.png">

I've followed all the steps as per [GitHub Flow Docs](https://docs.github.com/en/get-started/quickstart/github-flow), still I'm unable to see any "Feedback Pull Request" option.

Can you please help me out with this if I am missing something.

**Reply by heysujal** [reply]
try to click on that Contribute button and see if there is an option to make a PR

**Reply by ishaanbedi** [reply]
That will open a PR to the main branch only. Is that the way to make it a Feedback PR?

**Reply by heysujal** [reply]
Yes I think so.

**Reply by bdougie** [reply]
Yes, that is what is needed. There is no autogenerated PR. I have updated the upstream repo instructions and have removed the mention of the Feedback repo.

**Reply by yashladekar** [reply]
@bdougie  I have already finished my extension on my branch and generated a pull request; should I merge it now, before I start working on new features, or do I have to work on those first and then merge it?

**Reply by parmishh** [reply]
@bdougie I have pushed my work on a new branch I have finished it and want a review what are the next steps should I take?

**Comment by bdougie** [comment, marked as answer]
Sorry for the confusion

There has been some confusion on where the feedback PR is generated. When creating this assignment that was turned off. Instead you will need to manually make a PR to the main branch. 

If you have made changes to the main branch and not on a new branch, run the following in the command line on the main branch. 

Please feel free to mention me @bdougie if you have questions. 

```
git remote add upstream https://github.com/open-sauced-craftwork/browser-extension
git branch -m development
git push origin develop
git fetch upstream main
git checkout --track upstream/main
git push origin main -f // overwrites your current main branch with the original history.
```

After running these commands in succession you will be able to open the PR on your repo. That will allow me to review your code. 

Don’t forget to sign up to open sauced by [connecting your GitHub account](https://insights.opensauced.pizza/start) if you haven’t yet.

**Reply by vijaypotnuru** [reply]
can you giude  me to start assignment
